Feat/coffeechat 유저 CRUD + 시간 등록 기능 초기 구현

geonuu09 opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
커피챗 정보에 대한 초기 CRUD 기능 구현하였습니다. 
- userType 
  - GRADUATE : 수료자
  - PROFESSIONAL : 현업자
  - GENERAL : 일반유저
  
- jobType 
  - BACKEND, FRONTEND...(추후 추가)
  
- DayOfWeek
  - java 기본 제공 라이브러리 사용
  - MONDAY, THESDAY...

**AS-IS** 
- 커피챗 시간관리 기능 없음. 
- 커피챗 유저 등록만 있음. 
- application.porperties 

**TO-BE**
- 커피챗 시간 등록 구현 
- 커피챗 유저 등록 구현 
- application.yml 변경 
- record 타입 사용한 Dto 변경


- 코드 리뷰 후 변경 필요한 부분 수정하여 푸시하였습니다! 
  - 공통 메소드
  - deleted 필드 제거 및 로직 수정
### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [X] API 테스트 

- 커피챗 유저 등록
![image](https://github.com/user-attachments/assets/33e56389-f8ab-4419-923d-80e71f7fe3bf)

- 커피챗 시간 등록 
![image](https://github.com/user-attachments/assets/a31d7ec2-03bb-41f7-ba0b-a4dd62a06142)

- 내 커피챗 조회 
![image](https://github.com/user-attachments/assets/16885982-c564-41da-88cc-42529b9a5fe6)

- 내 커피챗 수정 
![image](https://github.com/user-attachments/assets/6e5d6081-f87f-4be9-837a-7791720212fb)

- 내 커피챗 삭제 
![image](https://github.com/user-attachments/assets/a9649692-11f6-42dd-a754-d1bf009c7529)

현재 커피챗 시간 등록 시 같은 유저가 같은 시간을 등록하는 것에 대해 중복 체크 및 예외처리가 없습니다. 논의 후 추가여부 결정해야할 것 같습니다!

geonuu09 committed:
feat : 커피챗 정보 CRUD 구현

- CRUD 초기 구현완료
- 유저 인증 추가 및 검색 API예정

geonuu09 committed:
Merge branch 'develop' of https://github.com/BootTalk-Service/boot-talk-back into feat/coffeechat

geonuu09 committed:
feat: 커피챗 유저 정보 등록 수정

geonuu09 committed:
feat: 커피챗 시간 관리 등록 기능

geonuu09 committed:
chore: application.yml 수정

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatInfoService.java:
```diff
@@ -0,0 +1,81 @@
+    @Transactional
+    public CoffeeChatInfoResponseDto deleteMyCoffeeChatInfo(Long userId) {
+
+        CoffeeChatInfo coffeeChatInfo = coffeeChatInfoRepository.findByUser_UserId(userId)
```
공통 사항으로 들어가는 로직을 메소드로 관리하는 것은 어떨까요??

geonuu09 replied:
네 반영해서 변경하도록 하겠습니다!

geonuu09 replied:
반영하였습니다!

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/service/CoffeeChatInfoService.java:
```diff
@@ -0,0 +1,81 @@
+    public CoffeeChatInfoResponseDto getMyCoffeeChatInfo(Long userId) {
+        CoffeeChatInfo coffeeChatInfo = coffeeChatInfoRepository.findByUser_UserId(userId)
+            .orElseThrow(() -> new CustomException(ErrorCode.USER_NOT_FOUND));
+
```
이미 내가 삭제 했던 Info와 새로 생성된 Info 둘다 있을 경우에는 어떤식으로 조회할지 판단을 안해도 되는지 알고 싶습니다!

geonuu09 replied:
deleted 필드 사용의 필요성이 없다고 판단되어 제거하도록 하겠습니다!

geonuu09 committed:
refactor : 커피챗 필드 수정에 따른 코드 수정
- deleted 필드와 관련된 코드 제거
- 레포지토리 delete 제거 추가

jooyeon0907 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatInfoRequestDto.java:
```diff
@@ -0,0 +1,24 @@
+) {
+
+    public void validate() {
+        if (introduction != null && introduction.length() > 1000) {
```
현재 1000이라는 값이 소개글 최대 길이를 나타내고 있는데, 
이를 상수로 관리하면 코드의 가독성과 유지 보수성이 향상될 것 같아요. 
조심스레 의견 내어봅니다!

geonuu09 replied:
아 이 부분에 대해 정책이 분명히 하지 않은 것 같아서 논의 후 결정하려고 대충 적었는데 잊고 있었습니다! 변경하도록 하겠습니다!

geonuu09 replied:
`@Size` 어노테이션을 사용하여 처리하였습니다. 기존에 validation 의존성을 사용하지 않았어서 사용을 못했던 것 같아요!

geonuu09 committed:
refactor: CoffeeChatInfo 조회 로직을 공통 메소드로 추출

- 사용자 ID로 CoffeeChatInfo를 조회하는 중복 코드를 getCoffeeChatInfoByUserId() 메소드로 추출

jooyeon0907 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/entity/CoffeeChatInfo.java:
```diff
@@ -0,0 +1,96 @@
+@Getter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
저는 원래 @NoArgsConstructor 만 사용했었는데,
기본 생성자의 접근 제어를 PROTETED 로 설정함으로써 더 안전한 설계를 고려하신 점 보고 배웁니다!

geonuu09 committed:
refactor: 소개글 길이 제한 값을 상수로 변경

- 소개글(introduction)의 최대 길이 제한을 상수 `MAX_INTRODUCTION_LENGTH`로 관리하도록 변경

geonuu09 committed:
refactor: 소개글 길이 검증 로직을 @Size 어노테이션으로 대체

- 기존에 `validate()` 메서드를 사용하여 `introduction` 필드의 길이를 검증하던 로직을 `@Size` 어노테이션을 사용하여 처리하도록 변경
- 소개글의 최대 길이를 1000자로 제한하고, 이를 초과하면 적절한 에러 메시지가 반환되도록 개선
